src/Symfony/Bundle/SecurityBundle/Security/FirewallMap.php

```diff
@@ -24,15 +24,19 @@
  */
 class FirewallMap implements FirewallMapInterface
 {
-    protected $container;
-    protected $map;
-    private $contexts;
+    /**
+     * @deprecated since version 3.3, to be removed in 4.0 alongside with magic methods below
+     */
+    private $map;
+    private $container;
+    private $contextCache;
+    private $contextMap;
 
-    public function __construct(ContainerInterface $container, array $map)
+    public function __construct(ContainerInterface $container, $contextMap)
     {
         $this->container = $container;
-        $this->map = $map;
-        $this->contexts = new \SplObjectStorage();
+        $this->contextCache = new \SplObjectStorage();
+        $this->contextMap = $contextMap;
```

Comment:
do we really need to change the name of these props? we'd better keep it as is for a small diff if we can